tests/auth/test_environment_variables_are_loaded_correctly.py

```diff
@@ -0,0 +1,248 @@
+from typing import Optional
+
+import pydantic
+import pytest
+
+from django_setup_configuration.model_utils import create_config_source_models
+from django_setup_configuration.models import ConfigurationModel
+from testapp.models import SomeModel
+
+
+class MyConfigModel(ConfigurationModel):
+    nested_obj: Optional[dict] = pydantic.Field()
+
+    class Meta:
+        django_model_refs = {SomeModel: ("foo",)}
+
+
+class MyListedConfigModel(ConfigurationModel):
+    items: list[MyConfigModel]
+
+
+@pytest.fixture()
+def valid_configuration_obj_with_value_from_env():
+    return {
+        "extra": "Extra vars at the root level are valid",
+        "config_enabled": True,
+        "the_namespace": {
+            "foo": {
+                "value_from": {
+                    "env": "FOO",
+                }
+            },
+            "nested_obj": {"nested_foo": "a nested string"},
+        },
+    }
```

Comment:
I think that this can be moved in the fixture below? I don't think it's used anywhere else?